casepro/backend/junebug.py

```diff
@@ -40,6 +46,75 @@ def get_addresses(self, uuid):
         return (
             a['address'] for a in addresses if a.get('address') is not None)
 
+    def get_identities(self, **params):
+        '''Get the list of identities filtered by the given kwargs.'''
+        url = '%s/api/v1/identities/?' % self.base_url
+
+        identities = self.get_paginated_response(
+            url, params=params)
+
+        # Users who opt to be forgotten from the system have their details
+        # stored as 'redacted'. We only want to return them if we are
+        # specifically looking for forgotten users.
+        return (
+            IdentityStoreContact(i) for i in identities if
+            i.get('details').get('name') != "redacted" or
+            'optout_type' in params
+        )
+
+
+class IdentityStoreContact(object):
+    """
+    Holds identity data for syncing
+    """
+    def __init__(self, json_data):
+        for k, v in json_data.items():
+            setattr(self, k, v)
+
+        # Languages in the identity store have the country code at the end
+        remote_language = json_data.get('details').get('preferred_language')
+        language, _, _ = remote_language.partition('_')
```

Comment:
We can just do `self.language` here to avoid having to set it below (since I don't think that we are using `language` anywhere else).

Also, what if there is no `preferred_language` in `details`? That would mean that we are doing a `None.partition` here, which would result in an error. Maybe this should be in an `if remote_language is not None:`, that way no language is set if the preferred language is not set.

As far as I can see from the identity store, only `addresses` and `default_address_type` are required in the `details` field.